code/entities/weapons/WeaponBase.cs

```diff
@@ -167,5 +184,25 @@ public override void CreateViewModel()
         ViewModelEntity.EnableViewmodelRendering = true;
         ViewModelEntity.SetModel(ViewModelPath);
     }
+
+    public void RemoveFromHolder()
+    {
+        if (!WeaponHolder.IsValid())
+            return;
+
+        WeaponHolder.Inventory.RemoveWeapon(this, false);
```

Comment:
This will break with slot based inventory changes, I will leave it up to you to decide which should merge first.